flytekit/image_spec/default_builder.py

```diff
@@ -114,26 +128,54 @@ def _is_flytekit(package: str) -> bool:
     return name == "flytekit"
 
 
-def create_docker_context(image_spec: ImageSpec, tmp_dir: Path):
-    """Populate tmp_dir with Dockerfile as specified by the `image_spec`."""
-    base_image = image_spec.base_image or "debian:bookworm-slim"
+def prepare_uv_lock_command(image_spec: ImageSpec, pip_install_args: List[str], tmp_dir: Path) -> str:
+    # uv sync is experimental, so our uv.lock support is also experimental
+    # the parameters we pass into install args could be different
+    warnings.warn("uv.lock support is experimental", UserWarning)
 
-    requirements = []
+    if image_spec.packages is not None:
+        msg = "uv.lock is not compatible with packages, please include all your dependencies in uv.lock"
```

Comment:
nit: should this mention that support for `uv.lock` files and packages is mutually exclusive instead?